src/lib/utils/asyncStorage.js

```diff
@@ -0,0 +1,45 @@
+import AsyncStorage from 'react-native'
+
+export default new class {
+  constructor(storageApi) {
+    this.storageApi = storageApi.AsyncStorage
+  }
+
+  async setItem(key, value) {
+    try {
```

Comment:
to keep consistence with original interface we should
1. rethrow errors instead of return
2. do it when getItem/setItem call fails only. If JSON.parse fails just return undefined

so:

```javascript
async setItem(key, value) {
  const stringified = JSON.stringify(value)
  
  await this.storageApi.setItem(key, stringified) // will re-throw storage error
}

async getItem(key) {
  const jsonValue = await this.storageApi.getItem(key) // will re-throw reading error

   if (jsonValue === null) { // if null is read just returning null
      return null
    }

  try {
    return JSON.parse(jsonValue) // trying to parse
  } catch {
    return // return undefined if parse fails
  }
}

```